R/layers.R

```diff
@@ -131,12 +131,14 @@ sass_layer_struct <- function(
 
   validate_attachments(file_attachments)
 
-  if (inherits(html_deps, "html_dependency")) {
+  if (is_dependency_maybe(html_deps)) {
     html_deps <- list(html_deps)
   }
   if (!is.null(html_deps)) {
-    is_dependency <- vapply(html_deps, inherits, logical(1), "html_dependency")
-    if (any(!is_dependency)) stop("html_deps must be a collection of htmltools::htmlDependency() objects", call. = FALSE)
+    is_dependency <- vapply(html_deps, is_dependency_maybe, logical(1))
```

Comment:
Let's move the `is_dependency_maybe()` check inside the `!is.null()` check.
And add another stop-check for non-list html_deps.  

Then we can keep the warning for unknown value types